homework-submission/react-hyf-client/src/Containers/Users.jsx

```diff
@@ -0,0 +1,109 @@
+import React, { useState, useEffect, useCallback } from 'react';
+import Button from '../components/Button';
+import User from '../components/User';
+import Region from '../components/Region';
+import { countries } from '../services/countries';
+import Spinner from '../components/Spinner';
+import ErrorModal from '../Error/ErrorModal.jsx';
+const Users = () => {
+  const [data, setData] = useState([]);
+  const [randomUser, setRandomUser] = useState([]);
+  const [isLoading, setIsLoading] = useState(true);
+  const [gender, setGender] = useState('random');
+  const [region, setRegion] = useState('random');
+  const [isShowing, setIsShowing] = useState(false);
+  const [changed, setChange] = useState(false);
+  const [error, setError] = useState();
+
+  const fetchData = useCallback(async () => {
+    try {
+      await fetch(`https://uinames.com/api/?amount=30&gender=${gender}&region=${region}&ext`)
+        .then(res => res.json())
+        .then(res => {
+          setData(res);
+          setIsLoading(false);
+        });
+    } catch (error) {
+      setError(error.message);
+    }
+  }, [gender, region]);
+
+  useEffect(() => {
+    fetchData();
+  }, [fetchData]);
+
+  const getUserHandler = useCallback(
+    e => {
+      e.preventDefault();
+      if (data.length > 0) {
+        if (e.keyCode === 32) {
+          const index = Math.floor(Math.random() * data.length);
+          const newData = [data[index]];
+          setRandomUser(newData);
+          setChange(true);
+        }
+      }
+      return null;
+    },
+    [data],
+  );
+
+  useEffect(() => {
+    window.addEventListener('keyup', getUserHandler);
+    // Remove event listeners on cleanup
+    return () => {
+      return window.addEventListener('keyup', getUserHandler);
+    };
+  }, [getUserHandler]);
+
+  const queryGenderHandler = gender => {
+    setGender(gender);
+  };
+  console.log('dd');
```

Comment:
Always remove `console.log`s when creating a pull request 🙃

Reply:
Thanks a lot for having time to review my homework!
I forgot to remove it!  😅